src/com/intellij/idea/plugin/hybris/project/view/HybrisProjectView.kt

```diff
@@ -76,19 +79,64 @@ open class HybrisProjectView(val project: Project) : TreeStructureProvider, Dumb
             .filter { isNodeVisible(parent, it) }
             .toMutableList()
 
-        if (parent is JunkProjectViewNode) {
+        if (parent is JunkProjectViewNode || parent is ExternalProjectViewNode) {
             return if (isCompactEmptyMiddleFoldersEnabled(settings)) compactEmptyMiddlePackages(parent, newChildren)
             else newChildren
         }
 
         if (parent is ExternalLibrariesNode) return modifyExternalLibrariesNodes(newChildren)
-
-        val childrenWithProcessedJunkFiles = processJunkFiles(newChildren, settings)
+        val externalModules = processExternalModules(parent, newChildren, settings)
+        val childrenWithProcessedJunkFiles = processJunkFiles(externalModules, settings)
 
         return if (isCompactEmptyMiddleFoldersEnabled(settings)) compactEmptyMiddlePackages(parent, childrenWithProcessedJunkFiles)
         else childrenWithProcessedJunkFiles
     }
 
+    private fun processExternalModules(
+        parent: AbstractTreeNode<*>,
+        children: Collection<AbstractTreeNode<*>>,
+        settings: ViewSettings?
+    ): Collection<AbstractTreeNode<*>> {
+        if (!hybrisApplicationSettings.groupExternalModules || isExternalModuleParent(parent)) return children
+
+        val otherNodes = mutableListOf<AbstractTreeNode<*>>()
+        val treeNodes = mutableListOf<AbstractTreeNode<*>>()
+
+        val projectService = ApplicationManager.getApplication().getService(HybrisProjectService::class.java)
+        for (child in children) {
+            if (child is PsiDirectoryNode) {
+                val virtualFile = child.virtualFile
+                    ?: continue
+                val file = VfsUtil.virtualToIoFile(virtualFile)
+
+                if (!projectService.isHybrisModule(file)
+                    && !projectService.isPlatformModule(file)
+                    && !projectService.isPlatformExtModule(file)
+                    && (projectService.isGradleModule(file)
+                        || projectService.isEclipseModule(file)
+                        || projectService.isGradleKtsModule(file)
+                        || projectService.isMavenModule(file))
+                ) {
+                    otherNodes.add(child)
+                } else {
+                    treeNodes.add(child)
+                }
+            } else {
+                treeNodes.add(child)
+            }
+        }
+        if (otherNodes.isNotEmpty()) {
+            treeNodes.add(ExternalProjectViewNode(project, otherNodes, settings))
```

Comment:
it looks like we will always group external modules, even if there is only 1 external module.
we can, for sure, group even 1 module, nothing bad in that.